open-broker-common/src/main/java/de/fraunhofer/iais/eis/ids/broker/core/common/persistence/ResourcePersistenceAndIndexing.java

```diff
@@ -176,6 +181,11 @@ public URI updated(Resource resource, URI connectorUri) throws IOException, Reje
             logger.info("Resource already exists. Removing");
             removeFromTriplestore(resource.getId(), connectorUri);
         }
+        //if a resource exists and its connectorUri is not null, ideaööy only resource should be removed here
+        if (resourceExists(resource.getId()) && !(connectorUri == null))   {
```

Comment:
This block doesn't make any sense, seeing that the "if" statement above already checks whether a resource exists and deletes it if it does. Here, you are repeating the check in a more restrictive way. I would assume that the "removeFromTriplestore" would just throw a "NotFound" exception, as we are already deleting the resource above